source/common/json/json_sanitizer.cc

```diff
@@ -0,0 +1,77 @@
+#include "source/common/json/json_sanitizer.h"
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+
+namespace Envoy {
+namespace Json {
+
+JsonSanitizer::JsonSanitizer() {
+  // Single-char escape sequences for common control characters.
+  auto control_char = [this](char escape, absl::string_view letter) {
+    char_escapes_[static_cast<uint32_t>(escape)] =
+        std::make_unique<std::string>(absl::StrCat("\\", letter));
+  };
```

Comment:
can we construct this `char_escapes_` array only once and shared it across all the `JsonSanitizer` instance?

Reply:
I was thinking there would only be one of these, instantiated by StatsHandler or as a CONSTRUCT_ON_FIRST_USE. JsonSanitizer otherwise has no state; only const methods.

Reply:
if we can make the char_escapes_ array static by the `CONSTRUCT_ON_FIRST_USE`, then `sanitize` can be a static method. This should make the method much easier to use.

Reply:
Yes we can, but that makes the fast-path a *lot* slower, because it needs to do an atomic test and construct on every access. There's nothing wrong with using CONSTRUCT_ON_FIRST_USE for the whole object but I think applications will want to save a reference to the JsonSanitizer object and use that in a fast loop, rather than force the atomic operation on every sanitization.

Reply:
> it needs to do an atomic test and construct on every access

I'm a bit confused about this point, why? Why can we just construct a static `char_escapes_` and ref it in our static `sanitize` method? The `char_escapes_` should only be constructed for once. 🤔 Is there something I've missed?

Reply:
I'm envisioning what needs to occur in when the compiler generates code for the code behind CONSTRUCT_ON_FIRST_USE. It the first use can come from any thread, so it needs to lazily, atomically, create the object. The perf results confirm this is not free. though I didn't look exactly at what gets generated.

Reply:
A normal static local variable is used for the CONSTRUCT_ON_FIRST_USE. The compile guarantees it's initialization is thread-safe. But l do don't know if there would be some additional overhead or not.

Okay, then I get it. I will also take a look at the compiler afterwards and test if there is additional overhead after local static variable initialization.

Reply:
I think if we verify later that CONSTRUCT_ON_FIRST_USE does not cause performance problems, we can consider modifying the implementation here again.

Reply:
I already verified it does cause performance problems, and moreover for the use-case I have in mind it's easy to instantiate one in a context that lives for the entire server, or even for the entire request.